src/cascadia/TerminalSettingsEditor/SettingContainer.cpp

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "SettingContainer.h"
+#include "SettingContainer.g.cpp"
+#include "LibraryResources.h"
+
+using namespace winrt::Windows::UI::Xaml;
+
+namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
+{
+    DependencyProperty SettingContainer::_HeaderProperty =
+        DependencyProperty::Register(
+            L"Header",
+            xaml_typename<IInspectable>(),
+            xaml_typename<Editor::SettingContainer>(),
+            PropertyMetadata{ nullptr });
+
+    DependencyProperty SettingContainer::_HelpTextProperty =
+        DependencyProperty::Register(
+            L"HelpText",
+            xaml_typename<hstring>(),
+            xaml_typename<Editor::SettingContainer>(),
+            PropertyMetadata{ box_value(L"") });
+
+    DependencyProperty SettingContainer::_HasSettingValueProperty =
+        DependencyProperty::Register(
+            L"HasSettingValue",
+            xaml_typename<bool>(),
+            xaml_typename<Editor::SettingContainer>(),
+            PropertyMetadata{ box_value(true), PropertyChangedCallback{ &SettingContainer::_OnHasSettingValueChanged } });

```

Comment:
remember to somehow make these only initialize on the first construction of one of your objects. A It's expensive to run all this winrt code the moment the TSE library is loaded (which is when Terminal boots up :|)